lib/src/extensions/line_stream.dart

```diff
@@ -181,22 +180,38 @@ extension LineStreamExtensions on Stream<String> {
   ///
   /// If [name] is passed, it's used as the name of the spawned script.
   ///
-  /// See also [LineStreamExtensions.xargs], which takes arguments from stdin
-  /// rather than from a string stream.
+  /// [Script.kill] closes the input stream. It returns `true` if the stream was
+  /// interrupted and the script exits with [Script.exitCode] `143`, or `false`
+  /// if the stream was already closed.
+  ///
+  /// The [callback] can't be interrupted by calling [kill], but the [onSignal]
+  /// callback allows capturing those signals so the callback may react
+  /// appropriately. Calling [kill] returns `true` even without an [onSignal]
+  /// callback if the stream is interrupted.
+  ///
+  /// See also `xargs` in `package:cli_script/cli_script.dart`, which takes
+  /// arguments from [stdin] rather than from this string stream.
   Script xargs(FutureOr<void> callback(List<String> args),
-      {int? maxArgs, String? name}) {
+      {int? maxArgs, String? name, void onSignal(ProcessSignal signal)?}) {
     if (maxArgs != null && maxArgs < 1) {
       throw RangeError.range(maxArgs, 1, null, 'maxArgs');
     }
 
-    return Script.capture((stdin) async {
-      if (maxArgs == null) {
-        await callback(await toList());
-      } else {
-        await for (var slice in slices(maxArgs)) {
-          await callback(slice);
-        }
-      }
-    }, name: name ?? 'xargs');
+    var completer = Completer<int>();
+    late StreamSubscription<void> subscription;
+    return Script.capture(
+        (_) async {
+          var chunks = maxArgs != null ? slices(maxArgs) : toList().asStream();
```

Comment:
`toList()` here won't actually cancel the underlying stream when the script is killed (something to add a test for!), because it just returns a `Future` which doesn't have any notion of cancelablity.

The more I think about this the more I think you were probably right to use `StreamCloser` in the first place—trying to actually manage an explicit subscription here is probably too complex. Sorry for the churn!

Reply:
hm I guess I could default to max int 64 🤔

Reply:
This is clever... but I think your original code with `StreamCloser` was probably still cleaner.